packages/portalnetwork/src/subprotocols/contentLookup.ts

```diff
@@ -101,55 +92,48 @@ export class ContentLookup {
-            if (!peer.hasContent) {
-              if (
-                !this.protocol.routingTable.contentKeyKnownToPeer(
-                  peer.nodeId,
-                  toHexString(this.contentKey)
-                )
-              ) {
-                // Only offer content if not already offered to this peer
-                this.protocol.sendOffer(peer.nodeId, [this.contentKey])
-              }
+          for (const peer of this.contacted) {
+            if (
+              !this.protocol.routingTable.contentKeyKnownToPeer(peer, toHexString(this.contentKey))
+            ) {
+              // Only offer content if not already offered to this peer
+              this.protocol.sendOffer(peer, [this.contentKey])
             }
-          })
+          }
           return res.value
         }
         case 2: {
           // findContent request returned ENRs of nodes closer to content
           this.logger(`received ${res.value.length} ENRs for closer nodes`)
-          res.value.forEach((enr) => {
+          for (const enr of res.value) {
             if (!finished) {
               const decodedEnr = ENR.decode(Buffer.from(enr as Uint8Array))
-              if (nodesAlreadyAsked.has(decodedEnr.nodeId)) {
-                return
-              }
-              const dist = distance(decodedEnr.nodeId, this.contentId)
-              if (this.lookupPeers.length === 0) {
-                // if no peers currently in lookup table, add to beginning of list
-                this.lookupPeers.push({ nodeId: decodedEnr.nodeId, distance: dist })
-              } else {
-                const index = this.lookupPeers.findIndex((peer) => peer.distance > dist)
-                if (index > -1) {
-                  // add peer to lookupPeer list if distance from content is less than at least one current lookupPeer
-                  this.lookupPeers.splice(index - 1, 0, {
-                    nodeId: decodedEnr.nodeId,
-                    distance: dist,
-                  })
-                } else {
-                  // if distance to content is greater than all other peers, add to end of lookupPeer list
-                  this.lookupPeers.push({ nodeId: decodedEnr.nodeId, distance: dist })
-                }
+              // Disregard if nodes have been previously contacted during this lookup,
+              // Or if nodes are currently being ignored for unresponsiveness.
+              if (
+                this.contacted.includes(decodedEnr.nodeId) ||
+                this.protocol.routingTable.isIgnored(decodedEnr.nodeId)
+              ) {
+                continue
               }
+              // Send a PING request to check liveness of any unknown nodes
               if (!this.protocol.routingTable.getValue(decodedEnr.nodeId)) {
-                this.protocol.routingTable.insertOrUpdate(decodedEnr, EntryStatus.Connected)
+                const ping = await this.protocol.sendPing(decodedEnr)
+                if (!ping) {
```

Comment:
If we get a peerId during a lookup and that peer doesn't respond on the initial ping, do you think we should still keep them in the routing table or go ahead and evict them?  Seems like if we don't get an initial response from a node on the first ping, we should go ahead and boot them, right?

Reply:
Yes, agreed.  That was the intention -- good catch :+1:

Reply:
Kind of wondering if we should bother with the 3 strikes thing at all, or just be savage about evicting nodes that don't PONG